### PR TITLE
Compute differentials, load csv as dataframes, tests

### PR DESCRIPTION
**Pull request by jsalbert**
- New functions in utils:
    * `results_to_dataframe(results, id='default_model', mode='average', round_decimals=3, show_id=True)` : Converts results format to pandas dataframe. 
    * `load_csv_to_dataframe(csv_paths)`: Reads 1 or more csv files and convert them to pandas dataframes.
    * `results_differential(dataframes, mode='average', round_decimals=4)`: Takes dataframes and return a dataframe with a nice visualization of the metrics differential. This function will be optimized in the future as right now we believe it is not the optimal way to access data positions. 

screen shot 2018-11-19 at 5 22 53 pm

screen shot 2018-11-19 at 4 05 41 pm

- Now `eval.show_results()` calls to `results_to_dataframe()`.

### CODE REVIEW

File: keras_eval/eval.py [hunk 1]
```diff
@@ -520,11 +520,17 @@ def plot_top_k_accuracy(self):
...
 
-    def show_results(self, mode='average', csv_path=None, round_decimals=3):
+    def show_results(self, mode='average', round_decimals=3, show_id=True):
         if self.results is None:
             raise ValueError('results parameter is None, please run a evaluation first')
 
-        return utils.show_results(self.results, self.id, mode, csv_path, round_decimals)
+        return utils.results_to_dataframe(self.results, self.id, mode, round_decimals, show_id)
+
+    def save_results(self, id, csv_path, mode='average', round_decimals=3, show_id=True):
```
[the remark this answers is not in the document]
Comment (jsalbert) [reply keras_eval/eval.py]: Yes, thanks for pointing that out

File: keras_eval/eval.py [hunk 2]
```diff
@@ -520,11 +520,40 @@ def plot_top_k_accuracy(self):
         metrics = self.results['average']['accuracy']
         visualizer.plot_concept_metrics(['all'], [metrics], 'Top-k', 'Accuracy')
 
-    def show_results(self, mode='average', csv_path=None, round_decimals=3):
+    def show_results(self, mode='average', round_decimals=3, show_id=True):
+        '''
+        Args:
+
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
```

File: keras_eval/eval.py [hunk 3]
```diff
@@ -520,11 +520,40 @@ def plot_top_k_accuracy(self):
...
+            mode: Mode of results. "average" will show the average metrics while "individual" will show metrics by class
+            round_decimals: Decimal position to round the numbers.
+            show_id: Show id in the first column.
+
+        Returns: Pandas dataframe with results.
+
+        '''
+        if self.results is None:
+            raise ValueError('results parameter is None, please run a evaluation first')
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
            raise ValueError('results parameter is None, please run an evaluation first')
```

File: keras_eval/eval.py [hunk 4]
```diff
@@ -520,11 +520,40 @@ def plot_top_k_accuracy(self):
...
+        Args:
+
+            mode: Mode of results. "average" will show the average metrics while "individual" will show metrics by class
+            round_decimals: Decimal position to round the numbers.
+            show_id: Show id in the first column.
+
+        Returns: Pandas dataframe with results.
+
+        '''
+        if self.results is None:
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
        if not self.results:
```

File: keras_eval/eval.py [hunk 5]
```diff
@@ -520,11 +520,40 @@ def plot_top_k_accuracy(self):
...
+            raise ValueError('results parameter is None, please run a evaluation first')
+
+        return utils.results_to_dataframe(self.results, self.id, mode, round_decimals, show_id)
+
+    def save_results(self, id, csv_path, mode='average', round_decimals=3, show_id=True):
+        '''
+
+        Args:
+
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
```

File: keras_eval/eval.py [hunk 6]
```diff
@@ -520,11 +520,40 @@ def plot_top_k_accuracy(self):
...
+            id: Name of the results evaluation
+            csv_path: If specified, results will be saved on that location
+            mode: Mode of results. "average" will show the average metrics while "individual" will show metrics by class
+            round_decimals: Decimal position to round the numbers.
+            show_id: Show id in the first column.
+
+        Returns: Nothing. Saves pandas dataframe on csv_path specified.
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
        Returns: Nothing. Saves Pandas dataframe on csv_path specified.
```

**Pull request merged**